publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -25,58 +32,84 @@
-    
+
     public static final String IDENTIFIER_MISMATCH_ERROR_MESSAGE = "Identifiers in path and in body, do not match";
+    private final TicketService ticketService;
     private final ResourceService resourceService;
     private final IdentityServiceClient identityServiceClient;
-    
+
     /**
      * Default constructor for MainHandler.
      */
     @JacocoGenerated
     public UpdatePublicationHandler() {
         this(new ResourceService(
-                AmazonDynamoDBClientBuilder.defaultClient(),
-                Clock.systemDefaultZone()),
-            new Environment(),
+                 AmazonDynamoDBClientBuilder.defaultClient(),
+                 Clock.systemDefaultZone()),
+             TicketService.defaultService(),
+             new Environment(),
              IdentityServiceClient.prepare());
     }
-    
+
     /**
      * Constructor for MainHandler.
      *
      * @param resourceService publicationService
      * @param environment     environment
      */
     public UpdatePublicationHandler(ResourceService resourceService,
+                                    TicketService ticketService,
                                     Environment environment,
                                     IdentityServiceClient identityServiceClient) {
         super(UpdatePublicationRequest.class, environment);
         this.resourceService = resourceService;
+        this.ticketService = ticketService;
         this.identityServiceClient = identityServiceClient;
     }
-    
+
     @Override
     protected PublicationResponse processInput(UpdatePublicationRequest input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
-        
+
         SortableIdentifier identifierInPath = RequestUtil.getIdentifier(requestInfo);
         validateRequest(identifierInPath, input);
         Publication existingPublication = fetchExistingPublication(requestInfo, identifierInPath);
         Publication publicationUpdate = input.generatePublicationUpdate(existingPublication);
+        createPublishingRequestOnFileUpdate(publicationUpdate);
         Publication updatedPublication = resourceService.updatePublication(publicationUpdate);
         return PublicationResponse.fromPublication(updatedPublication);
     }
-    
+
+    private void createPublishingRequestOnFileUpdate(Publication publicationUpdate) throws ApiGatewayException {
+        if (containsNewFiles(publicationUpdate)) {
+            TicketEntry.requestNewTicket(publicationUpdate, PublishingRequestCase.class)
+                .persistNewTicket(ticketService);
+        }
+    }
+
     @Override
     protected Integer getSuccessStatusCode(UpdatePublicationRequest input, PublicationResponse output) {
         return HttpStatus.SC_OK;
     }
-    
+
+    private boolean containsNewFiles(Publication publicationUpdate) {
+        return !getUnpublishedFiles(publicationUpdate).isEmpty();
```

Comment:
return publicationUpdate.getAssociatedArtifacts().stream().anyMatch(this::isUnpublishedFile);

Reply:
done, takk